nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py

```diff
@@ -1432,25 +1498,55 @@ def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0, log_
                 speech_mask,
                 _,
                 cross_attention_prior,
-                text_limits,
+                text_limits,  # [start of question token, question token len) in [0, enc_mask.size(1))
                 lang,
                 question_texts,
             ) = batch
 
-            dec_input = dec_input_raw * 1  # (B, 8, T)
+            batch_size = virtual_tokens.size(0)
+            dec_input = dec_input_raw * 1  # (B, 8, T)  # TODO @xueyang: apply clone() method bypasses this unnecessary computation.
             dec_input_mask = dec_input_mask_raw * 1  # (B, T)
             dec_input_mask[:, :] = 1  # Does not really matter
             output_token_list = []
 
             end_indices = {}
             # pad dec_input (B, 8, T) to 1000 timesteps
             max_inference_timesteps = self.cfg.get('max_inference_timesteps', 2000)
+            # TODO @xueyang: potential bug when max_inference_timesteps < dec_input.shape[2], then dec_input is clipped.
             dec_input = torch.nn.functional.pad(dec_input, (0, max_inference_timesteps - dec_input.shape[2]), value=0)
             dec_input[:, :, self.decoder_context_len + 1:].zero_()
+            # TODO @xueyang: why not just declare torch.ones(dec_input_raw.size(0), max_inference_timesteps)?
             dec_input_mask = torch.nn.functional.pad(
                 dec_input_mask, (0, max_inference_timesteps - dec_input_mask.shape[1]), value=1
             )
 
+            # question unconditioned case
+            if self.question_guidance_scale != 1.0:
```

Comment:
Can you add an inference cfg parameter or option to the predict step?

Reply:
sure thing. Actually, I've already introduced `apply_text_cfg` and `apply_audio_cfg` params in my text and/or audio cfg updates. Something like below. Will refactor this PR a little bit to combine both text and audio cfg together.

```suggestion
            # text unconditioned case
            if self.apply_text_cfg:
                ....
            
            # audio unconditioned case
            if self.apply_audio_cfg:
                ...
```

Reply:
code snippet is here: https://gitlab-master.nvidia.com/xueyang/nemo/-/blob/xueyang/speechllm_tts_2402_cfg_audio/nemo/collections/tts/models/speechllm/megatron_t5_speechllm_model.py?ref_type=heads#L1541-1578